File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        seen = {0: 0}
        total = 0
        for i in range(len(nums)):
            total += nums[i]
            r = total % k
            if r not in seen:
                seen[r] = i + 1
            elif seen[r] < i:
                return True
        return False

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        Map<Integer, Integer> seen = new HashMap<>();
        seen.put(0, 0);
        
        int total = 0;
        int r;
        for(int i = 0; i < nums.length; i++){
            total += nums[i];
            r = total % k;
            if(!seen.containsKey(r)){
                seen.put(r, i + 1);
            } else if(seen.get(r) < i){
                return true;
            }
        }
        
        return false;
    }
}